utils/csv2yaml.py

```diff
@@ -0,0 +1,176 @@
+import argparse
+import pandas as pd
+from pathlib import Path
+import yaml
+
+
+SUPPORTED_TYPES = {"text": "string", "integer": "integer", "date": "date", "float": "float",
+                   "boolean": "boolean"}
+CONVERT_TYPES = {"integer": int}
+
+class YamlDumper(yaml.Dumper):
+    def write_line_break(self, data=None):
+        super(YamlDumper, self).write_line_break(data)
+        if len(self.indents) == 2:  # This checks if we're at the top level of the document
+            super(YamlDumper, self).write_line_break()  # Insert an additional line break
+
+
+def _create_class_entry(df_row):
+    """Create a class entry for a given row in the dataframe"""
+    variable_names = df_row["LinkML Slot or Attribute Name"]
+    class_dict = {}
+    if df_row["Definition_Class"]:
+        class_dict["description"] = df_row["Definition_Class"]
+    if df_row["Inheritance: is_a"]:
+        class_dict["is_a"] = df_row["Inheritance: is_a"]
+    if df_row["Inheritance: mixin"]:
+        class_dict["mixins"] = df_row["Inheritance: mixin"].split(",")
+    if df_row["Subsets_Class"]:
+        subsets_l = [el.strip() for el in df_row["Subsets_Class"].split(",")]
+        class_dict["in_subset"] = subsets_l
+
+    if not pd.isna(df_row["Class Name"]):
+        class_dict["attributes"] = \
+            dict((k, {"description": d.strip()}) for k,d in zip(variable_names, df_row["Definition"]))
+        for ii, dtp in enumerate(df_row["Data Type"]):
+            # if data type is valueset, then we need to check for permissible values
+            if dtp.lower() == "valueset":
+                if df_row['Permissible Values'][ii]:
+                    class_dict["attributes"][variable_names[ii]]["range"] = df_row['Permissible Values'][ii].title().replace(" ", "")
+                else:
+                    raise Exception("No Permissible Value")
```

Comment:
this has to be specific to the user seeing the errors and should contain more general info. also use a more specific exception than a generic one.

Reply:
yes, sorry, exceptions were not finished yet